pkgs/development/libraries/libnats-c/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub
+, cmake, protobuf, protobufc
+, libsodium, openssl
+}:
+
+stdenv.mkDerivation rec {
+  pname   = "libnats";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner  = "nats-io";
+    repo   = "nats.c";
+    rev    = "refs/tags/v${version}";
```

Comment:
How does this work? `fetchFromGitHub` will use fetchzip if you didn't use `fetchSubmodules`, so this would make it fetch `"https://github.com/nats-io/nats.c/archive/refs/tags/v${version}.tar.gz"` not `https://github.com/nats-io/nats.c/archive/v2.1.0.tar.gz`.